### DISCUSSION TITLE
Build Fails

### DISCUSSION THREAD

**Discussion by waltercolindres** [vercel/next.js#14820] [Help] [answered]
Site works and runs locally in dev mode. When I build to preview locally or push up the build fails on vercel.

Package Json:
```
{
  "scripts": {
    "dev": "next dev",
    "build": "next build"
  },
  "dependencies": {
    "@contentful/rich-text-from-markdown": "^14.1.1",
    "@contentful/rich-text-links": "^14.1.1",
    "@contentful/rich-text-react-renderer": "^14.1.1",
    "babel-cli": "^6.26.0",
    "contentful": "^7.14.4",
    "next": "^9.4.0",
    "normalize.css": "^8.0.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "sass": "^1.26.9"
  }
}

```

```10:30:44.531  
Compiled successfully.
10:30:44.532  
Automatically optimizing pages...
10:30:44.817  
> Build error occurred
10:30:44.818  
[String: 'The resource could not be found.'] {
10:30:44.818  
  type: 'Object',
10:30:44.818  
  stack: null,
10:30:44.818  
  sys: { type: 'Error', id: 'NotFound' },
10:30:44.819  
  message: 'The resource could not be found.',
10:30:44.819  
  requestId: 'ced292f5-7621-4546-b21c-f186939af458'
10:30:44.819  
}
10:30:44.841  
error Command failed with exit code 1.
10:30:44.841  
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
10:30:44.851  
Error: Command "yarn run build" exited with 1```

```
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'build' ]
2 info using npm@6.9.0
3 info using node@v10.16.3
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle @~prebuild: @
6 info lifecycle @~build: @
7 verbose lifecycle @~build: unsafe-perm in lifecycle true
8 verbose lifecycle @~build: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/XXXXX/node_modules/.bin:/Applications/google-cloud-sdk/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
9 verbose lifecycle @~build: CWD: /Users/XXXXX
10 silly lifecycle @~build: Args: [ '-c', 'next build' ]
11 silly lifecycle @~build: Returned: code: 1  signal: null
12 info lifecycle @~build: Failed to exec build script
13 verbose stack Error: @ build: `next build`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:198:13)
13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:198:13)
13 verbose stack     at maybeClose (internal/child_process.js:982:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
14 verbose pkgid @
15 verbose cwd /XXXXXXX
16 verbose Darwin 19.4.0
17 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
18 verbose node v10.16.3
19 verbose npm  v6.9.0
20 error code ELIFECYCLE
21 error errno 1
22 error @ build: `next build`
22 error Exit status 1
23 error Failed at the @ build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by IRediTOTO** [comment]
Are you running on a VPS ? Did you try build on vercel?

**Reply by waltercolindres** [reply]
I tried on vercel. Here was the output:

`10:30:21.627  
Cloning completed in 951ms
10:30:21.627  
Analyzing source code...
10:30:23.419  
Installing build runtime...
10:30:23.804  
Build runtime installed: 385.161ms
10:30:24.214  
Looking up build cache...
10:30:24.278  
Build cache not found
10:30:24.700  
Installing dependencies...
10:30:24.913  
yarn install v1.22.4
10:30:24.927  
warning package.json: No license field
10:30:24.972  
warning No license field
10:30:24.973  
[1/4] Resolving packages...
10:30:25.312  
[2/4] Fetching packages...
10:30:32.752  
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
10:30:32.752  
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
10:30:32.756  
info fsevents@2.1.3: The platform "linux" is incompatible with this module.
10:30:32.757  
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
10:30:32.760  
[3/4] Linking dependencies...
10:30:35.926  
[4/4] Building fresh packages...
10:30:36.493  
Done in 11.58s.
10:30:36.516  
Running "yarn run build"
10:30:36.771  
yarn run v1.22.4
10:30:36.784  
warning package.json: No license field
10:30:36.793  
$ next build
10:30:37.359  
info  - Loaded env from /vercel/3ba8ac57/.env.production
10:30:37.361  
Creating an optimized production build...
10:30:37.391  
Attention: Next.js now collects completely anonymous telemetry regarding usage.
10:30:37.391  
This information is used to shape Next.js' roadmap and prioritize features.
10:30:37.392  
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
10:30:37.392  
https://nextjs.org/telemetry
10:30:44.531  
Compiled successfully.
10:30:44.532  
Automatically optimizing pages...
10:30:44.817  
> Build error occurred
10:30:44.818  
[String: 'The resource could not be found.'] {
10:30:44.818  
  type: 'Object',
10:30:44.818  
  stack: null,
10:30:44.818  
  sys: { type: 'Error', id: 'NotFound' },
10:30:44.819  
  message: 'The resource could not be found.',
10:30:44.819  
  requestId: 'ced292f5-7621-4546-b21c-f186939af458'
10:30:44.819  
}
10:30:44.841  
error Command failed with exit code 1.
10:30:44.841  
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
10:30:44.851  
Error: Command "yarn run build" exited with 1
10:30:46.621  
Done with "package.json"`

**Reply by IRediTOTO** [reply]
Never meet this `[String: 'The resource could not be found.'] {`
I think problem come from one of you npm packge. Sorry can't help you.

**Comment by waltercolindres** [comment, marked as answer]
So I was missing the env keys on vercel. oddly still doesn't compile locally but gets further now on vercel.
